gc/base/MemoryPoolLargeObjects.cpp

```diff
@@ -538,11 +538,18 @@ MM_MemoryPoolLargeObjects::allocateTLH(MM_EnvironmentBase* env, MM_AllocateDescr
 void*
 MM_MemoryPoolLargeObjects::findFreeEntryEndingAtAddr(MM_EnvironmentBase* env, void* addr)
 {
+	void *freeEntry = NULL;
 	if (addr >= _currentLOABase) {
-		return _memoryPoolLargeObjects->findFreeEntryEndingAtAddr(env, addr);
+		if (getCurrentLOASize() == getApproximateFreeLOAMemorySize()) {
+			Assert_MM_true(((void*)(((uintptr_t)addr) - _loaSize)) == _currentLOABase);
```

Comment:
reorder things so to use additions rather subtraction (here and one assert below)